bmfdec.c

```diff
@@ -279,11 +279,16 @@ int main() {
     fprintf(stderr, "Failed to read data: %s\n", strerror(EFBIG));
     return 1;
   }
-  if (lin <= 16 || pin[0] != 0x424D4F46 || pin[1] != 0x01 || pin[2] != (uint32_t)lin-16 || pin[3] > sizeof(pout)) {
+  if (lin <= 16 || pin[0] != 0x424D4F46 || pin[1] != 0x01 || pin[2] != (uint32_t)lin-16) {
     fprintf(stderr, "Invalid input\n");
     return 1;
   }
   lout = pin[3];
+  char *pout = malloc(lout);
```

Comment:
This is insecure. Malicious input can cause OOM. It is needed to check for some upper bound limit.

Probably better would be to move static allocation from stack to static global allocation (moving buffer out of main function) and increase its limit. If 64K is not enough then increasing limit to 256K should be fine for now.

Reply:
Thanks, I have moved the buffers to static storage duration and bumped up the size as you suggest. Rebased and pushed.

Reply:
perfect!